hibernate-core/src/main/java/org/hibernate/query/sqm/internal/QuerySqmImpl.java

```diff
@@ -817,14 +817,18 @@ public LockOptions getLockOptions() {
 
 	@Override
 	public SqmQueryImplementor<R> setLockOptions(LockOptions lockOptions) {
-		verifySelect();
+		if ( lockOptions.getLockMode() != LockMode.NONE ) {
+			verifySelect();
```

Comment:
```suggestion
			// This is required by the JPA spec
			verifySelect();
```

Reply:
I can add it, but there is similar comment within `verifySelect()` already:
```
	protected void verifySelect() {
		try {
			SqmUtil.verifyIsSelectStatement( getSqmStatement(), hql );
		}
		catch (IllegalQueryOperationException e) {
			// per JPA
			throw new IllegalStateException( "Expecting a SELECT query : `" + hql + "`", e );
		}
	}
```